### PR TITLE
Update the PR template

### PR DESCRIPTION
**Pull request by nathaniel-may**
### Description

The goal is to minimize the items in the checklist so it is easier to abide by, and to add clear reminders for opening docs tickets and getting explicit approval from Product/DX for interface changes.

Let's discuss the proposed change here, and if we agree to merge I'll open identical PRs against the adapter repos.

### Changes
- Removed changie checklist item because it is a required branch protection and automation responds with instructive PR comments
- Removed CLA checklist item because it is a required branch protection and automation responds with instructive PR comments
- Removed docs checklist item in favor of adding a place to link the open docs tickets at the top
- Added Product/DX approval checklist item to remind contributors to call out interface changes specifically
- Split "Description" header into "Problem" and "Solution".

### CODE REVIEW

**Comment by mikealfare** [comment]
Just to play devil's advocate, how do know for sure what an interface change is? And is this end user interface? Adapter maintainer interface? etc.

File: .github/pull_request_template.md
Comment (jtcohen6) [inline .github/pull_request_template.md:20]: Does this include all of:
- dbt developers, writing code in their projects
- dbt developers, invoking dbt commands/params on the CLI
- consumers of dbt metadata (logs + JSON artifacts)
- maintainers of adapter plugins

**Review by jtcohen6 — COMMENTED** [review]
I'm supportive of us being more conscious of interface changes, especially when incidental / unintended. An additional prompt in the PR description is a good start.

What do you think about adding these to the "Checklist" below?
- **Product/DX approval:** PRs with significant interface changes require explicit approval from Product or DX. That could be approving the PR, or having written / signed off on an earlier design doc for which this PR is the implementation.
- **Docs:** "I have opened an issue to add/update docs" - and that issue details the interface change explicitly, as something that will need to be called out in [the "What to know before upgrading" section](https://docs.getdbt.com/guides/migration/versions/upgrading-to-v1.5#what-to-know-before-upgrading) of the version migration guide

**Comment by nathaniel-may** [comment]
@jtcohen6 @dbeatty10 

We can use checklist bullets instead of sections if we like that better. We could also incorporate the interface list you included in the above comment to be more explicit and specific.

> Product/DX approval: PRs with significant interface changes require explicit approval from Product or DX. That could be approving the PR, or having written / signed off on an earlier design doc for which this PR is the implementation.

could instead become

Product/DX approval: This PR has no interface changes (e.g.  macros, cli, logs, json artifacts, config files, adapter interface) or this PR has already received feedback and approval from Product or DX.

**Comment by dbeatty10** [comment]
@nathaniel-may I like your suggestions!

Either checklist bullets _or sections would work equally fine from a DX perspective. Since engineers are the intended audience, we should go with whatever would serve as the best aid to them.

**Comment by jtcohen6** [comment]
> Either checklist bullets _or_ sections would work equally fine from a DX [+ Product] perspective. Since engineers are the intended audience, we should go with whatever would serve as the best aid to them.

Seconded!

**Comment by nathaniel-may** [comment]
I'm expanding the scope of this PR to include more template changes after speaking to @martynydbt @dbeatty10 and @Mathyoub.

File: .github/pull_request_template.md
Comment (martynydbt) [inline .github/pull_request_template.md:19]: Does the pr fail if they didn't sign it?

**Review by martynydbt — APPROVED** [review]
Looks great to me, thank you for following up on this!

File: .github/pull_request_template.md
Comment (dbeatty10) [inline .github/pull_request_template.md:22]: Would it be useful to preserve an easy-to-click link somewhere?

https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose

File: .github/pull_request_template.md
Comment (dbeatty10) [inline .github/pull_request_template.md:34]: Is there a reasonable way to indicate which of the two mutually exclusive choices was applicable?

When I see a checked box that has two mutually exclusive items ("A or B"), I wonder -- okay, which one was it?

File: .github/pull_request_template.md
Comment (nathaniel-may) [reply .github/pull_request_template.md:22]: Definitely! I have it in the description at the top right now.

File: .github/pull_request_template.md
Comment (nathaniel-may) [reply .github/pull_request_template.md:34]: I pushed a new commit that attempts to address this. I can revert it if we don't like it though.

Comment (nathaniel-may) [reply .github/pull_request_template.md]: I'm finding it difficult to format it nicely in markdown without it looking like a huge checklist now. How important is it for this distinction to be in the checklist to you?

Comment (nathaniel-may) [reply .github/pull_request_template.md]: Here's the [commit](https://github.com/dbt-labs/dbt-core/blob/7ad6628222af66d0efcd343e6920e018cbd8cbdb/.github/pull_request_template.md) with my best attempt at modeling the split.

File: .github/pull_request_template.md
Comment (dbeatty10) [reply .github/pull_request_template.md:22]: Ah, I see it in the commented section now 👀

Personally, I'd find it harder to open a docs issue with this version of the PR versus the current one.

But it's totally up to the engineers. I support whichever they find easiest.

File: .github/pull_request_template.md
Comment (dbeatty10) [reply .github/pull_request_template.md]: I'd prefer for it to be clear, especially as it relates to interface changes. Agreed that a short checklist looks a lot better than a long one.

If our template were a little verbose like your attempt, how onerous would it be to edit it down after opening the PR? i.e., like the "Initial" vs. "Edited" below? Or how painful would it be for folks to see the longer checklist?

## Initial

```markdown
### Checklist

- [ ] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me  
- [ ] I have run this code in development and it appears to resolve the stated issue  
Tests - choose one:  
- [ ] This PR includes tests  
- [ ] Tests are not required or are not relevant for this PR  
Interfaces - choose one:  
- [ ] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc)  
- [ ] This PR has already received feedback and approval from Product or DX.  
```

## Edited

```markdown
### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me  
- [x] I have run this code in development and it appears to resolve the stated issue  
- [x] Tests are not required or are not relevant for this PR  
- [x] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc)  
```

File: .github/pull_request_template.md
Comment (martynydbt) [inline .github/pull_request_template.md:9]: nit: "opened for this PR" instead of issue

File: .github/pull_request_template.md
Comment (martynydbt) [reply .github/pull_request_template.md:22]: Can you share what feels harder? I do love bullets for simplicity over sentences in paragraphs, but that's probably not what you mean.

Comment (nathaniel-may) [reply .github/pull_request_template.md:22]: I'm gonna guess it's the ability to click a link. I made the top "docs" text a link. Does that help, @dbeatty10?

File: .github/pull_request_template.md
Comment (nathaniel-may) [reply .github/pull_request_template.md]: This is a totally reasonable idea, but it is a process change from how the checklist has functioned before 🤔

File: .github/pull_request_template.md
Comment (dbeatty10) [reply .github/pull_request_template.md:22]: Yep, that is what I meant and that looks easier @nathaniel-may 👍

**Comment by nathaniel-may** [comment]
I'm going to merge as it is now, and we can update it again if we decide to automate the doc portion as described in https://github.com/dbt-labs/actions/issues/111

**Pull request merged**
